packages/json-signing/README.md

```diff
@@ -0,0 +1,36 @@
+# JSON Data Signing
```

Comment:
There seems to be a standardized solution to the pattern you are suggesting, in a framework called [JOSE](https://jose.readthedocs.io/en/latest/) and I see at least one npm [package](https://www.npmjs.com/package/node-jose) that implements it. Perhaps it is worth evaluating.

Reply:
Absolutely!

Reply:
So the RFCs for JOSE do not cover the use of `ed25519` or schnorr keys - and I've not seen their use in NPM package either.

And then there's this here: https://paragonie.com/blog/2017/03/jwt-json-web-tokens-is-bad-standard-that-everyone-should-avoid

Reply:
However, going from that latter link, we might use https://paseto.io/ - specifically https://github.com/sjudson/paseto.js which in `v2` uses `ed25519`.

The standard also allows encrypted messages, but we really only care about signing here. I'll try the library out a little, and if it works as expected, will change the data signing section to use this scheme.

Reply:
Paseto looks promising, but doesn't seem to deliver at this point in time. It's not necessarily the spec's or the implementation's fault, though: it seems that sodium, which is used under the hood, requires 64-Byte secret keys for signing messages, while polkadot provides us with 32-Byte secret keys.

If we use the polkadot key to derive a 64-Byte key, then public keys won't match any longer, and we'd have to manage yet another public key for the same identity, probably on chain.

It seems simpler to use the polkadot key's sign/verify capabilities directly, in a scheme as proposed in this PR. However, now I'll want to verify that this works before committing to it.

Reply:
Yup, that works. So I'm inclined to leave the spec largely as-is, with references to why we're not using JOSE or Paseto. WDYT?

Reply:
Added that section for your reviewing pleasure, @mnaamani :)

Reply:
Yeah stretching a 32-byte key and the added overhead of managing one more public key is not worth it.
So lets stick with our own spec and implementation.

Reply:
I think it will be nice to have signing working for acropolis, but if time doesn't permit, it can always go into rome.